### PR TITLE
Enable fetch polyfill with web client

### PR DESCRIPTION
**Pull request by timja**
thanks to @basil for finding this out in https://github.com/jenkinsci/jenkins/pull/7727#discussion_r1155349075

Required for https://issues.jenkins.io/browse/JENKINS-70906

I've tested this by enabling the option in the failing test in the above pull request and it makes the test pass

### CODE REVIEW

**Review by basil — APPROVED** [review]
Nice job fixing this in both `HudsonTestCase` and `JenkinsRule`. You are a pro!

---

This means that any PCT tests in plugins running against a core that contains any of

- https://github.com/jenkinsci/jenkins/pull/7727
- https://github.com/jenkinsci/jenkins/pull/7728
- https://github.com/jenkinsci/jenkins/pull/7731
- https://github.com/jenkinsci/jenkins/pull/7781

_may_ result in HtmlUnit failures if they are also not running HtmlUnit 2.59.0 or later (i.e., `plugin-pom` 4.38 or later) _and_ this PR. At the time of this writing, no plugins are running the code in this PR.

To ensure `jenkinsci/bom` is not impacted, I have requested `jenkinsci/bom` test runs before further usages of the Fetch API are introduced in Jenkins core.

I am CC'ing @jtnord as CloudBees internal PCT testing _may_ be impacted.

**Comment by basil** [comment]
Also note that e.g. `maven-plugin` has two direct calls to `new WebClient()` that are not being addressed by this PR. They ought to be converted to calls to `createWebClient()`.

**Comment by timja** [comment]
> Also note that e.g. `maven-plugin` has two direct calls to `new WebClient()` that are not being addressed by this PR. They ought to be converted to calls to `createWebClient()`.

https://github.com/jenkinsci/maven-plugin/pull/308

File: src/main/java/org/jvnet/hudson/test/HudsonTestCase.java [hunk 1]
```diff
@@ -1396,7 +1396,9 @@ protected Object writeReplace() {
      * This is to assist Groovy test clients who are incapable of instantiating the inner classes properly.
```
Comment (daniel-beck) [inline src/main/java/org/jvnet/hudson/test/HudsonTestCase.java:1396]: Newly misleading Javadoc?

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java [hunk 2]
```diff
@@ -2214,7 +2214,9 @@ protected Object writeReplace() {
      * This is to assist Groovy test clients who are incapable of instantiating the inner classes properly.
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/org/jvnet/hudson/test/JenkinsRule.java:2214]: Is it though? Why would people have been calling it from Java in almost every test class if that was the case

[the remark this answers is not in the document]
Comment (basil) [reply src/main/java/org/jvnet/hudson/test/JenkinsRule.java:2214]: :laughing: The comment was flawed before and it's still flawed.

**Pull request merged**